stdlib/auth/src/main/java/org/ballerinalang/auth/ldap/nativeimpl/Authenticate.java

```diff
@@ -0,0 +1,140 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.auth.ldap.nativeimpl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.ballerinalang.auth.ldap.util.CommonLDAPConfiguration;
+import org.ballerinalang.auth.ldap.util.LDAPConnectionContext;
+import org.ballerinalang.auth.ldap.util.LDAPConstants;
+import org.ballerinalang.auth.ldap.util.LDAPUtils;
+import org.ballerinalang.auth.ldap.util.Secret;
+import org.ballerinalang.auth.ldap.util.UserStoreException;
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.bre.bvm.BlockingNativeCallableUnit;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.model.values.BBoolean;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+import javax.naming.AuthenticationException;
+import javax.naming.NamingException;
+import javax.naming.ldap.LdapContext;
+
+/**
+ * Ballerina function to authenticate users with LDAP user store.
+ */
+@BallerinaFunction(
+        orgName = "ballerina", packageName = "auth",
+        functionName = "LDAPAuthStoreProvider.doAuthenticate",
+        args = {@Argument(name = "username", type = TypeKind.STRING),
+                @Argument(name = "password", type = TypeKind.STRING)
+        },
+        returnType = {@ReturnType(type = TypeKind.BOOLEAN)},
+        isPublic = true)
+public class Authenticate extends BlockingNativeCallableUnit {
+
+    private static final Log LOG = LogFactory.getLog(Authenticate.class);
+    private LDAPConnectionContext connectionSource;
+    private CommonLDAPConfiguration ldapConfiguration;
+
+    @Override
+    public void execute(Context context) {
+
+        BMap<String, BValue> authStore = ((BMap<String, BValue>) context.getRefArgument(0));
+        connectionSource = (LDAPConnectionContext) authStore.getNativeData(LDAPConstants.LDAP_CONNECTION_SOURCE);
+        ldapConfiguration = (CommonLDAPConfiguration) authStore.getNativeData(LDAPConstants.LDAP_CONFIGURATION);
+
+        String userName = context.getStringArgument(0);
+        Object credential = context.getStringArgument(1);
+
+        if (userName == null || credential == null) {
+            context.setReturnValues(new BBoolean(false));
+            throw new BallerinaException("username or credential value is null.");
+        }
+
+        userName = userName.trim();
+        Secret credentialObj = null;
+
+        try {
+            credentialObj = Secret.getSecret(credential);
+            if (userName.equals("") || credentialObj.isEmpty()) {
```

Comment:
Shall we check `userName..isEmpty()` here.

Reply:
+1 to use isEmpty and credentialObj cannot be null